src/audio/codec_adapter/codec/generic.c

```diff
@@ -251,3 +251,30 @@ int codec_prepare(struct comp_dev *dev)
 end:
 	return ret;
 }
+
+int codec_process(struct comp_dev *dev)
+{
+	int ret;
+	struct comp_data *cd = comp_get_drvdata(dev);
+	uint32_t codec_id = cd->ca_config.codec_id;
+	struct codec_data *codec = &cd->codec;
+
+	comp_dbg(dev, "codec_process() start");
+
+	if (cd->codec.state < CODEC_PREPARED) {
+		comp_err(dev, "codec_prepare(): wrong state of codec %x, state %d",
+			 cd->ca_config.codec_id, codec->state);
+		return -EPERM;
+	}
+
+	ret = codec->call->process(dev);
+	if (ret) {
```

Comment:
I feel that instead of ``codec->call->process`` we should have ``codec->ops->process``